### ISSUE TITLE
Linter Warnings: DO use curly braces for all flow control structures

### ISSUE THREAD

**Issue by WanderingFire2001** [xsahil03x/super_enum#34]
The generated code sometimes provokes the analyzer (under pedantic settings) to issue the warning: "DO use curly braces for all flow control structures".

It can be fixed either by generating braces when you have an if with a single line body (these single line bodies currently appear when you throw).

Or faster, you could take the approach that built_value does and just suppress warnings at the file level by generating the following line at the bottom of each .g.dart file:

```// ignore_for_file: curly_braces_in_flow_control_structures```

**Details:**
It happens *sometimes* where the generated code includes throws clauses that aren't surrounded by braces, e.g.:
```dart
//ignore: missing_return
  FutureOr<R> when<R>(
      {@required FutureOr<R> Function(UpdateFilter) updateFilter,
      @required FutureOr<R> Function(UpdateFeatures) updateFeatures}) {
    assert(() {
      if (updateFilter == null || updateFeatures == null)
        throw 'check for all possible cases';
      return true;
    }());
    ....
```
Here, it happens because ```throw 'check for all possible cases';``` is formatted alone on a line.

It only causes linter warnings sometimes, because oddly, if the formatter keeps the throw on the same line as the ```if``` or an ```||``` as shown below, there is no linter warning.

```dart
      if (updateFilter == null || ... || ...
          || updateFeatures == null) throw 'check for all possible cases';
```

**Comment by xsahil03x** [comment]
@WanderingFire2001 thanks for mentioning this issue. We are currently waiting on releasing some new features [#32 , #33] and will fix this as soon as those got merged. :smile:

**Comment by WanderingFire2001** [comment]
Thanks! I added a List field a couple days ago and am doing the inefficient ```nonGenericList.map(...).toList()``` into a ```List<MyType>``` until the fix comes through.

I'll have to think through the implications of the wrapper on the other one, but it sounds useful.

**Comment by astralstriker** [comment]
Fixed in #33
